src/app/dashboard/Missions/DashboardMissions.jsx

```diff
@@ -133,15 +133,24 @@ const PageButtons = ({ classes, pageView, setPageView }) => {
 
 const DashboardMissions = ({ inDone, inPlanning, inProgress, inProposed }) => {
   const classes = useStyles();
-  const all = { inProposed, inPlanning, inProgress, inDone };
 
   const [pageView, setPageView] = useState(pageViews.list);
-  let filtered = inProposed;
+  const [selectedMission, setSelectedMission] = useState(null);
 
+  const all = { inProposed, inPlanning, inProgress, inDone };
   const viewFromUrl = _.getQueryParam("view");
-  filtered = all[viewFromUrl];
+  const filtered = all[viewFromUrl] || inProposed;
+
+  const selectedMissionId = _.getQueryParam("missionId");
```

Comment:
I’ll have to take a look at our router set up more in-depth later, or not from my phone,  but react router should at least provide this to us in props with something like props.match.params.missionId

Reply:
Yeh I think our use of react router could def be better